### PR TITLE
HCAP-1208 | Multiselect on multiple pages

### PR DESCRIPTION
**Pull request by taraepp**
https://freshworks.atlassian.net/jira/software/c/projects/MOH/boards/244?modal=detail&selectedIssue=HCAP-1208

Decided to scope the multiselect to the current page, so that:
- selecting all selects all items on page
- changing page deselects selection
- the select all checkbox is displayed as checked when all the items on the page are selected

### CODE REVIEW

File: client/src/components/generic/Table.js [hunk 1]
```diff
@@ -262,8 +263,8 @@ export const Table = ({
                 <Checkbox
                   color='primary'
                   disabled={isLoading || rowsCount === 0}
-                  indeterminate={selectedRows.length > 0 && selectedRows.length < rowsCount}
-                  checked={rowsCount > 0 && selectedRows.length === rowsCount}
+                  indeterminate={selectedRows.length > 0 && selectedRows.length < rows.length} //
```
Comment (alva-fresh) [inline client/src/components/generic/Table.js:266]: extra `//` here?

Comment (taraepp) [reply client/src/components/generic/Table.js]: Fixed!

File: client/src/components/generic/Table.js [hunk 2]
```diff
@@ -262,8 +263,8 @@ export const Table = ({
                 <Checkbox
                   color='primary'
                   disabled={isLoading || rowsCount === 0}
-                  indeterminate={selectedRows.length > 0 && selectedRows.length < rowsCount}
-                  checked={rowsCount > 0 && selectedRows.length === rowsCount}
+                  indeterminate={selectedRows.length > 0 && selectedRows.length < rows.length}
+                  checked={rowsCount > 0 && selectedRows.length === rows.length}
```
Comment (alva-fresh) [inline client/src/components/generic/Table.js:267]: We probably can completely replace `rowsCount` with `rows.length`; and the remove it

**Review by fwpushan — COMMENTED** [review]
Please address the issue raised by @alva-fresh

File: client/src/components/generic/Table.js (hunk 2 above)
Comment (taraepp) [reply client/src/components/generic/Table.js]: I renamed rowsCount to totalRowsCount (it's used for pagination, can't get rid of it completely) and made a new variable rowsOnPage = rows.length.

**Review by alva-fresh — APPROVED** [review]

**Pull request merged**